avail/src/main/kotlin/avail/resolver/ModuleRootResolver.kt

```diff
@@ -361,7 +393,8 @@ constructor(
 		withReference: (ResolverReference)->Unit,
 		failureHandler: (ErrorCode, Throwable?) -> Unit)
 	{
-		val qname = qualifiedName.replace(availExtension, "")
+		val extension = availModuleFileExtension(qualifiedName)
+		val qname = qualifiedName.replace(extension, "")
```

Comment:
This would remove the extension string if it occurred in the middle of the string as well.  You should use `qualifiedName.removeSuffix(extension)`.